plugins/c9.vfs.server/download.js

```diff
@@ -113,10 +113,8 @@ define(function(require, exports, module) {
                 paths.forEach(function(path) {
                     if (!path) return;
                     path = Path.relative(cwd, path);
-                    // tar misinterprets the Windows path separator as an escape sequence, so use forward slash.
-                    if (Path.sep === '\\') {
-                        path = path.replace(/\\/g, '/');
-                    }
+                    // Single quote the path to escape unusual characters, and manually escape single quotes.
+                    path = "'" + path.replace(/'/, "'\\''") + "'";
```

Comment:
this does not work on linux either

run
```js
mkdir -p 0-ws/x/y/z/
mkdir -p 0-ws/a
echo 1 > 0-ws/x/y/z/z-file
echo 2 > 0-ws/x/y/y-file
echo 3 > 0-ws/x/x-file
echo 4 > 0-ws/root-file
echo 5 > "0-ws/root'quote"
find 0-ws
```
to create file like
```txt
0-ws
0-ws/root'quote
0-ws/x
0-ws/x/y
0-ws/x/y/z
0-ws/x/y/z/z-file
0-ws/x/y/y-file
0-ws/x/x-file
0-ws/root-file
0-ws/a
```

then try to download `0-ws/x/y/z/z-file` and `0-ws/root-file`. that doesn't work for both zip and tar, because node itself quotes the arguments.